src/sensai/vector_model.py

```diff
@@ -15,10 +15,35 @@
 T = TypeVar('T')
 
 
-class PredictorModel(ABC):
+class _IdentityFG(FeatureGenerator):
```

Comment:
What exactly is the reason for these classes? Why did you think it wasn't appropriate to use None-checks (not all of which were removed) instead?
None-checks being removed breaks backward compatibility (and loses convenience), because it would no longer be OK to pass None to the respective with* method (which can be very convenient for flexibly configured model construction). Granted, the possibility to pass None is not properly type annotated, but neither are your changes (e.g. regarding Y in this PR). It really is annoying to have to add Optional, isn't it?

Reply:
Yeah, I was unsure about that myself. It felt good not to have None checks everywhere in the RuleBasedModel 
(fit if not none, generate if not None and so on). We also have not-None input and output DTF transformers, so this change felt natural. I was thinking about backwards compatibility but I considered that it would be rather a nuisance. Do you perform such checks in your code at the moment?

Reply:
Concerning the incorrect type annotations, I need to check. Maybe you can give a more specific example? For kwargs with default None the Optional thing is inferred automatically

Reply:
Btw, I did allow passing None in the with* methods

Reply:
There were a few None checks in VectorModel but then I needed a few more in the RuleBased one and we might need more for further abstractions. The added complexity ist arguable, I would say - for me the code in fit and predict looks significantly neater without the additional None checks. I think that this code which will be read much more often than the builder methods setting the fgen or target transformer or these two trivial classes. Also, it is in line with having _inputTransformerChain and _outputTransformerChain not being None even when they are trivial. So overall, the tradeoff of having two private classes vs None checks seemed beneficial to me